CameraBrowser/CameraBrowserSample/ViewController.cs

```diff
@@ -0,0 +1,238 @@
+			DeviceBrowser.BrowsedDeviceTypeMask = ICBrowsedDeviceType.Local | ICBrowsedDeviceType.Remote | ICBrowsedDeviceType.Camera;
+			DeviceBrowser.Start();
+		}
+
+		[Export("observeValueForKeyPath:ofObject:change:context:")]
+		private void ObserveValueForKeyPath (NSString keyPath, NSArrayController ofObject, NSDictionary change, IntPtr context)
+		{
+			if (keyPath == "selectedObjects" && ofObject == MediaFilesController)
+			{
+				WillChangeValue("canDelete");
+				WillChangeValue("canDownload");
+				DidChangeValue("canDelete");
+				DidChangeValue("canDownload");
+
+			}
+		}
+
+		[Export("canDelete")]
+		public bool CanDelete ()
+		{
+			bool can = false;
+			InvokeOnMainThread(() => {
+				var selectedFiles = MediaFilesController.SelectedObjects;
+				foreach (ICCameraFile f in selectedFiles)
+				{
+					if (!f.Locked)
+					{
+						can = true;
+						break;
+					}
+				}
+			});
+			return can;
+		}
+
+		[Export("canDownload")]
+		public bool CanDownload()
+		{
+			var res = false;
+			InvokeOnMainThread(() => {
+				res = MediaFilesController.SelectedObjects is not null ? MediaFilesController.SelectedObjects.Length > 0 : false;
+			} );
+			return res;
+		}
+
+		[Export("selectedCamera")]
+		public ICCameraDevice? SelectedCamera
+		{
+			get
+			{
+				var selectedObjects = CamerasController.SelectedObjects;
+				if (selectedObjects is null || selectedObjects.Length == 0)
+					return null;
+				return selectedObjects[0] as ICCameraDevice;
+			}
+
+		}
+
+		[Export("downloadFiles:")]
+		public void downloadFiles (ICCameraFile[] files)
+		{
+			NSString filePath = new NSString("~/Pictures");
+			var filePath1 = filePath.ExpandTildeInPath();
+			string[] fileUrl = new string[] { filePath1 };
+
+			var objects = new[] { NSUrl.CreateFileUrl(fileUrl) };
+			var keys = new[] { ICCameraDownloadOptionKeys.DownloadsDirectoryUrl };
+			var dict = new NSDictionary<NSString, NSObject>(keys, objects);
+			foreach (ICCameraFile f in files)
+			{
+				f?.Device?.RequestDownloadFile(f, dict, didDownloadFileDel);
+			}
+		}
+
+		public void didDownloadFileDel(ICCameraFile f, NSError err, NSDictionary options)
+		{
+			Console.WriteLine("didDownloadFile called with:");
+			Console.WriteLine($"file: {f}");
+			Console.WriteLine($"error: {err}");
+			Console.WriteLine($"options: {options}");
+		}
+
+		[Export("readFiles:")]
+		public void readFiles (ICCameraFile[] files)
+		{
+			foreach(ICCameraFile f in files)
+			{
+				f?.Device?.RequestReadDataFromFile(f, 0, f.FileSize, didReadDataDel);
+			}
+		}
+
+		public void didReadDataDel(NSData data, ICCameraFile file, NSError err)
+		{
+			Console.WriteLine("didReadData called with:");
+			Console.WriteLine($"data: {data}");
+			Console.WriteLine($"file: {file}");
+			Console.WriteLine($"error: {err}");
+		}
+
+		[Export("openCamera")]
+		public void openCamera()
+		{
+			SelectedCamera?.RequestOpenSession();
+		}
+
+		// IICDeviceBrowserDelegate Impl
+
+		[Export("deviceBrowser:didAddDevice:moreComing:")]
```

Comment:
You shouldn't need the Export attribute on any method we implement from the interfaces this class implements ("IICDeviceBrowserDelegate, IICCameraDeviceDelegate, IICCameraDeviceDownloadDelegate, IICDeviceDelegate"), and I believe that's the rest of the methods in this class. Can you try removing these Export attributes and seeing everything still works as expected?